sympy/integrals/integrals.py

```diff
@@ -1243,6 +1242,144 @@ def _sage_(self):
                                     hold=True)
         return f
 
+    def cauchy_principal_value(self, sym, lower_limit=-oo, upper_limit=oo):
+        """
+        Compute the Cauchy Principal Value of the definite integral of a real function in the given interval
+        on the real axis.
+        In mathematics, the Cauchy principal value, is a method for assigning values to certain improper
+        integrals which would otherwise be undefined.
+        The Cauchy principal value of a finite integral of a function f about a point c with a <= c <= b is given by
+                b               c-r                       b
+              /                  /                      /
+             /                  /                      /
+             |                  |                      |
+             |                  |                      |
+        p.v. | f(x)dx  = lim    | f(x)dx   +   lim     |
+             |          r -> 0+ |             r -> 0+  |
+             |                  |                      |
+             /                  /                     /
+            /                  /                     /
+          a                 a                     c+r
+
+        Examples
+        ========
+
+        >>> from sympy import Dummy, symbols, integrate, limit, oo
+        >>> from sympy.integrals.integrals import Integral
+        >>> from sympy.calculus.singularities import singularities
+        >>> x = symbols('x', real = True)
+
+        >>> g = x + 1
+        >>> Integral(g).cauchy_principal_value(x)
+        oo
+
+        >>> f = 1/(x**3)
+        >>> singularities(f,x)
+        {0}
+
+        >>> Integral(f).cauchy_principal_value(x)
+        0
+
+        >>> Integral(f).cauchy_principal_value(x, -oo, 0)
+        -oo
+
+        >>> Integral(f).cauchy_principal_value(x, 0, oo)
+        oo
+
+        >>> Integral(f).cauchy_principal_value(x, 1, 2)
+        3/8
+
+        >>> Integral(f).cauchy_principal_value(x,-2,-1) + Integral(f).cauchy_principal_value(x,1,2)
+        0
+
+        >>> Integral(f).cauchy_principal_value(x, 0.4, 3)
+        3.06944444444444
+
+        >>> Integral(f).cauchy_principal_value(x,-3,-0.4) + Integral(f).cauchy_principal_value(x, 0.4, 3)
+        0
+
+
+        See also
+        ========
+
+        [1] https://en.wikipedia.org/wiki/Cauchy_principal_value
+        [2] http://mathworld.wolfram.com/CauchyPrincipalValue.html
+        """
+        from sympy.calculus import singularities
+
+        self.sym = sym
```

Comment:
These should not be added to `self`. It is an instance of `Basic` and should be immutable. Instead, these could be taken as local variables: `x, a, b = self.limits[0]`. (There should be exactly one item in `self.limits` and that should be a triple.)

Reply:
@jksuom 

Just let me know how you want the user to type in the command for a function
1. with limits `-oo to oo`
2. some other limits with some other variable ( if function is in variable `x` then sym is variable `y`)

I'll modify my function accordingly.

Reply:
If the integration variable is `y`, then the limits are written `(y, -oo, oo)` (or `(y, a, b)` for some other limits).